jax/random.py

```diff
@@ -374,20 +374,52 @@ def uniform(key: jnp.ndarray,
     dtype: optional, a float dtype for the returned values (default float64 if
       jax_enable_x64 is true, otherwise float32).
     minval: optional, a minimum (inclusive) value broadcast-compatible with shape for the range (default 0).
-    maxval: optional, a maximum (exclusive) value broadcast-compatible with shape for the range (default 1).
+    maxval: optional, a maximum (exclusive) value broadcast-compatible with shape for the range (default 1 if not complex, otherwise 1+1j).
 
   Returns:
     A random array with the specified shape and dtype.
   """
-  if not dtypes.issubdtype(dtype, np.floating):
+  if minval is None:
+    minval = _uniform_minval_default(dtype)
+  if maxval is None:
+    maxval = _uniform_maxval_default(dtype)
+
+  if not (jnp.issubdtype(dtype, np.floating) or jnp.issubdtype(dtype, np.complexfloating)):
```

Comment:
np.inexact covers both floating and complexfloating.

See: https://numpy.org/doc/stable/reference/arrays.scalars.html